gitbutler-ui/src/lib/components/HunkLine.svelte

```diff
@@ -1,9 +1,12 @@
 <script lang="ts">
 	import { create } from '$lib/components/Differ/CodeHighlighter';
+	import { SETTINGS_CONTEXT, type SettingsStore } from '$lib/settings/userSettings';
 	import { SectionType } from '$lib/utils/fileSections';
-	import { createEventDispatcher } from 'svelte';
+	import { createEventDispatcher, getContext } from 'svelte';
 	import type { Line } from '$lib/utils/fileSections';
 
+	const userSettings = getContext(SETTINGS_CONTEXT) as SettingsStore;
```

Comment:
This is great, but instead of getting this context here each time per line, how about moving it to `HunkViewer` and passing in the tab size as a prop to this component?

Reply:
ah yes, that makes much more sense :p

Reply:
Sorry, I have to ask.. Would it make even more sense to set this as a property on the hunk viewer level? The `HunkLine` component currently takes quite primitive properties, and can be instantiated without context dependencies.

Reply:
Potentially, but at that point should `project` also be passed down? Would they be gotten at the `FileCard` level and prop-drilled down?

Reply:
You're right.. and then we're just creating more props. Let's go with what you have right now!